compiler/luci/lang/include/luci/IR/CircleShapeSignature.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_IR_SHAPE_SIGNATURE_H__
+#define __LUCI_IR_SHAPE_SIGNATURE_H__
+
+#include <stdint.h>
+#include <vector>
+
+namespace luci
+{
+
+class ShapeSignature
+{
+public:
+  ShapeSignature() = default;
+
+  ShapeSignature(const std::vector<int32_t> &shape_signature)
+  {
+    _shape_signature = shape_signature;
+  }
+
+  const std::vector<int32_t> as_vector() const { return _shape_signature; }
+
+  const int32_t &dim(uint32_t d) const { return _shape_signature.at(d); }
```

Comment:
`const int32_t` vs `int32_t` in return type... do we need to add `const` for this?

Reply:
At first, I was not sure so I referenced following code...
https://github.com/Samsung/ONE/blob/92f5e69b4a8ac45a051e3f62768915931b38376f/compiler/loco/include/loco/IR/NodeMixins.h#L67-L68

I think that we may need to control the modification authority.

Reply:
`const int32_t &dim()` as `int32_t` is just a number, not a class or struct, I think it would be same for `const int32_t dim()`
For this I'm not sure `const` is usefull for the caller. just `int32_t dim()` wouldn't be better for usage?